app/views/meal_plans/show.html.erb

```diff
@@ -27,7 +27,7 @@
     <%= link_to '+ Add All Ingredients to Grocery List',
                 shopping_list_item_builders_path(
                   shopping_list_id: ShoppingList.default(current_user).id,
-                  meal_plan_id: @meal_plan.id
+                  ingredient_ids: @ingredient_set.values.flatten.pluck(:id)
```

Comment:
Rather than mucking with what you're already doing with `IngredientSet`, this will just grab the ingredients from that existing instance variable. It's not very pretty, and could use some refactoring, so let me know if you would prefer something else.

Reply:
I think i'd like to see a MealPlanHelper method here:
```ruby
def meal_plan_ingredient_ids(ingredient_set)
  ingredient_set.values.flatten.pluck(:id)
end
```